### PR TITLE
LPS-176188 Add delete action to filter

### PR DESCRIPTION
**Pull request by thektan**
Story: https://issues.liferay.com/browse/LPS-176188
Subtask: https://issues.liferay.com/browse/LPS-183987

### What is the goal of this PR?
- Adds "Delete" action in the Filters tab table

### What does it look like?
https://github.com/liferay-frontend/liferay-portal/assets/4496722/7b3da3f3-4187-4e80-89fd-1b48a2b8869b

### CODE REVIEW

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Filters.tsx [hunk 1]
```diff
@@ -359,6 +359,29 @@ function Filters({fdsView, fdsViewsURL, namespace}: IProps) {
 			),
 		});
 
+	const handleDelete = async ({item}: {item: Filter}) => {
+		const response = await fetch(
+			`${
+				item.type === 'date-time'
+					? API_URL.FDS_DATE_FILTERS
+					: API_URL.FDS_DYNAMIC_FILTERS
+			}/${item.id}`,
```
Comment (bryceosterhaus) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Filters.tsx:368]: nit: for readability I'd put this as a variable

```js
const url = `${item.type === 'date-time' ? API_URL.FDS_DATE_FILTERS : API_URL.FDS_DYNAMIC_FILTERS}/${item.id}`;

const response = await fetch(url  //...
```

**Review by bryceosterhaus — COMMENTED** [review]
Looks good!

I do think we will probably need a "confirm" type of step before actually deleting, otherwise it'd be easy to delete on accident.

**Comment by markocikos** [comment]
> I do think we will probably need a "confirm" type of step before actually deleting

Yep. Please see how we implemented this for [deletion of FDSEntries](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSEntries.tsx#L744-L795).

**Comment by thektan** [comment]
@bryceosterhaus @markocikos A confirmation message wasn't included in the mockups, so I'll double-check with Uge if that's something we want to add.

**Comment by thektan** [comment]
Uge updated the mockups to include a confirmation modal, so I'll add that in.

**Comment by thektan** [comment]
Added in the confirmation message. Tested rebasing off of https://github.com/liferay-frontend/liferay-portal/pull/3323 and it didn't have any conflicts.

**Comment by thektan** [comment]
ci:test:sf

**Comment by thektan** [comment]
ci:test:relevant

**Comment by thektan** [comment]
I'll need to add another commit to fix the interface `IFilter` naming, so this can wait for https://github.com/liferay-frontend/liferay-portal/pull/3323 to be merged first.

**Review by markocikos — APPROVED** [review]
LGTM! :sparkles:

**Comment by markocikos** [comment]
ci:forward

**Pull request closed without merging**